### ISSUE TITLE
head_dim 256 prefill fails to launch with cudaErrorInvalidValue on SM75/SM70 (SharedStorageQKVO smem overflow)

### ISSUE THREAD

**Issue by HuChundong** [flashinfer-ai/flashinfer#3528]
### Summary

`HEAD_DIM 256` prefill kernels fail to launch on Turing/Volta (SM75/SM70) with
`cudaErrorInvalidValue` ("invalid argument"). The prefill `SharedStorageQKVO` struct exceeds the
64 KiB opt-in shared-memory cap (`cudaDevAttrMaxSharedMemoryPerBlockOptin`) on these arches, so
`cudaFuncSetAttribute(kernel, cudaFuncAttributeMaxDynamicSharedMemorySize, ...)` is rejected and
the kernel never runs.

### Root cause

`SharedStorageQKVO` unconditionally reserved 1-element placeholder arrays for the NVFP4
scale-factor buffers (`k_sf_smem` / `v_sf_smem`) and the FP8 "repack" buffer (`kv_smem_repack`)
even for KV dtypes that never touch them. After `alignas(16)` each placeholder occupies 16 B,
inflating `sizeof(SharedStorageQKVO)` by 48 B (FP16) / 32 B (FP8).

For `HEAD_DIM 256` the q/k/v union already fills the entire 64 KiB opt-in budget, so the extra
padding pushes the struct past 65536 B:

| KV dtype | `sizeof(SharedStorageQKVO)` @ HEAD_DIM 256 | 64 KiB cap |
| -------- | ------------------------------------------ | ---------- |
| FP16     | 65584                                       | exceeds    |
| FP8      | 65568                                       | exceeds    |
| NVFP4    | 50192                                       | fits       |

On SM80+ the opt-in cap is larger (≥ 100 KiB), so the overflow is invisible there and not
exercised by upstream CI; the failure is specific to SM75/SM70.

### Reproduction

- Hardware: NVIDIA RTX 2080 Ti (`sm_75`)
- Any `head_dim = 256` FP16 or FP8-KV prefill, e.g. `single_prefill_with_kv_cache` /
  `BatchPrefillWithPagedKVCache` (originally observed via vLLM `BatchPrefillWithPagedKVCache`).

```python
import torch, flashinfer
S, Skv, H, D = 128, 512, 4, 256  # head_dim 256 -> SM75 smem-overflow path
q = torch.randn(S, H, D, device="cuda", dtype=torch.float16)
k = torch.randn(Skv, H, D, device="cuda", dtype=torch.float16)
v = torch.randn(Skv, H, D, device="cuda", dtype=torch.float16)
o = flashinfer.single_prefill_with_kv_cache(q, k, v)  # launch fails: invalid argument on sm_75
```

Observed error:

```
Check failed: (status == cudaSuccess) ... kernel launch failed, error: invalid argument
```

### Expected behavior

`head_dim 256` prefill should launch and produce correct results on SM75/SM70, as it does on
SM80+.

### Proposed fix

Move the conditional buffers into empty-base mixins so that, for dtypes that don't use them, they
contribute 0 bytes via the empty base optimization instead of a padded placeholder. This brings
FP16/FP8 back to exactly 65536 B at `HEAD_DIM 256` (NVFP4 unchanged), with no behavior change for
dtypes that do use the buffers and no change on SM90+.

PR: #3526

### Environment

- FlashInfer: `main`
- GPU: RTX 2080 Ti (`sm_75`, Turing); also affects SM70 (Volta)
- CUDA: 12.8
